pkg/controllers/status/cluster_status_controller.go

```diff
@@ -288,17 +290,18 @@ func (c *ClusterStatusController) updateStatusIfNeeded(cluster *clusterv1alpha1.
 	return controllerruntime.Result{RequeueAfter: c.ClusterStatusUpdateFrequency.Duration}, nil
 }
 
-func (c *ClusterStatusController) initializeGenericInformerManagerForCluster(clusterClient *util.ClusterClient) {
+func (c *ClusterStatusController) initializeGenericInformerManagerForCluster(clusterClient *util.ClusterClient) error {
 	if c.GenericInformerManager.IsManagerExist(clusterClient.ClusterName) {
-		return
+		return nil
 	}
 
 	dynamicClient, err := c.ClusterDynamicClientSetFunc(clusterClient.ClusterName, c.Client)
 	if err != nil {
-		klog.Errorf("Failed to build dynamic cluster client for cluster %s.", clusterClient.ClusterName)
-		return
+		klog.Errorf("Failed to build dynamic cluster client for cluster %s: %v", clusterClient.ClusterName, err)
+		return err
```

Comment:
ClusterDynamicClientSetFunc could return an error, we shouldn't ignore it, as it will cause the resource module to not work(informer not created)